mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/runtime/impl/ProvidedStringIdentifierMapping.java

```diff
@@ -41,12 +56,12 @@ public String getIdentifier(Object providedId, Supplier<?> entityProvider) {
 
```

Comment:
You should also change `getIdentifier(Object)` above. Just remove `String.valueOf()` and return `providedId` directly.
Please keep the "if" block: it's necessary.